cube: New build flags produce warnings

The CMakeBuildList.txt is being updated to report more warnings and to report warnings as errors.
The following issues were found in vkcube and vkcubepp:
```
/home/runner/work/Vulkan-Tools/Vulkan-Tools/cube/cube.c:1177:50: error: comparison of integers of different signs: 'uint32_t' (aka 'unsigned int') and 'int' [-Werror,-Wsign-compare]
        if (surfCapabilities.currentExtent.width != demo->width || surfCapabilities.currentExtent.height != demo->height) {
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~
/home/runner/work/Vulkan-Tools/Vulkan-Tools/cube/cube.c:1177:106: error: comparison of integers of different signs: 'uint32_t' (aka 'unsigned int') and 'int' [-Werror,-Wsign-compare]
        if (surfCapabilities.currentExtent.width != demo->width || surfCapabilities.currentExtent.height != demo->height) {
                                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~
2 errors generated.
```
```
D:\a\Vulkan-Tools\Vulkan-Tools\cube\cube.cpp(812): error C2220: warning treated as error - no 'object' file generated [D:\a\Vulkan-Tools\Vulkan-Tools\build\cube\vkcubepp.vcxproj]
D:\a\Vulkan-Tools\Vulkan-Tools\cube\cube.cpp(812): warning C4456: declaration of 'result' hides previous local declaration [D:\a\Vulkan-Tools\Vulkan-Tools\build\cube\vkcubepp.vcxproj]
  D:\a\Vulkan-Tools\Vulkan-Tools\cube\cube.cpp(734): note: see declaration of 'result'
```

--- a/cube/cube.c
+++ b/cube/cube.c
@@ -1174,7 +1174,8 @@ static void demo_draw(struct demo *demo) {
         VkSurfaceCapabilitiesKHR surfCapabilities;
         err = demo->fpGetPhysicalDeviceSurfaceCapabilitiesKHR(demo->gpu, demo->surface, &surfCapabilities);
         assert(!err);
-        if (surfCapabilities.currentExtent.width != demo->width || surfCapabilities.currentExtent.height != demo->height) {
+        if (surfCapabilities.currentExtent.width != (uint32_t)demo->width ||
+            surfCapabilities.currentExtent.height != (uint32_t)demo->height) {
             demo_resize(demo);
         }
     } else if (err == VK_ERROR_SURFACE_LOST_KHR) {

--- a/cube/cube.cpp
+++ b/cube/cube.cpp
@@ -809,7 +809,7 @@ void Demo::draw() {
     } else if (result == vk::Result::eSuboptimalKHR) {
         // SUBOPTIMAL could be due to resize
         vk::SurfaceCapabilitiesKHR surfCapabilities;
-        auto result = gpu.getSurfaceCapabilitiesKHR(surface, &surfCapabilities);
+        result = gpu.getSurfaceCapabilitiesKHR(surface, &surfCapabilities);
         VERIFY(result == vk::Result::eSuccess);
         if (surfCapabilities.currentExtent.width != width || surfCapabilities.currentExtent.height != height) {
             resize();

